cmd/metacache-walk.go

```diff
@@ -120,7 +120,9 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 		if contextCanceled(ctx) {
 			return ctx.Err()
 		}
+		s.walkMu.Lock()
 		entries, err := s.ListDir(ctx, opts.Bucket, current, -1)
+		s.walkMu.Unlock()
```

Comment:
Should we only lock ListDir() instead of locking reading metadata since that may serialize a lot of calls that can be independent after ListDir() @klauspost ?

Reply:
The reads will be on the same disk on all cases, so not sure why we wouldn't include reading metadata.

For disks with a single ongoing listing, there will be close to no change since uncontested mutexes are fast and it doesn't operate concurrently. 

Again, it is to limit the potential impact of many concurrent listing requests. They will of course not all hit the same disks, but the disks that do should remain more responsive to non-listing requests.